src/table/__integ__/table-interaction-metrics.test.ts

```diff
@@ -0,0 +1,148 @@
+      );
+
+    await testFn({ page, wrapper, getInteractionMetrics, waitForLoadingFinished });
+  });
+};
+const delay = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+test(
+  'has no metrics when the table does not load anything',
+  setupTest(async ({ getInteractionMetrics }) => {
+    expect(await getInteractionMetrics()).toHaveLength(0);
+  })
+);
+
+test(
+  'emits a metric when the user filters',
+  setupTest(async ({ page, wrapper, getInteractionMetrics, waitForLoadingFinished }) => {
+    await page.click(wrapper.findFilterSlot().findTextFilter().toSelector());
+    await page.keys('238-1');
+    await waitForLoadingFinished();
+
+    const metrics = await getInteractionMetrics();
+    expect(metrics).toHaveLength(1);
+    expect(metrics[0]).toEqual(
+      expect.objectContaining({
+        userAction: 'filter',
+        componentIdentifier: 'Instances',
+        instanceIdentifier: 'the-instances-table',
+        noOfResourcesInTable: 19,
+      })
+    );
+    expect(metrics[0].interactionTime).toBeGreaterThanOrEqual(500);
+    expect(metrics[0].interactionTime).toBeLessThanOrEqual(2000);
+  })
+);
+
+test(
+  'emits a metric when the user paginates',
+  setupTest(async ({ page, wrapper, getInteractionMetrics, waitForLoadingFinished }) => {
+    await page.click(wrapper.findPagination().findPageNumberByIndex(2).toSelector());
+    await waitForLoadingFinished();
+
+    const metrics = await getInteractionMetrics();
+    expect(metrics).toHaveLength(1);
+    expect(metrics[0]).toEqual(
+      expect.objectContaining({
+        userAction: 'pagination',
+        componentIdentifier: 'Instances',
+        instanceIdentifier: 'the-instances-table',
+        noOfResourcesInTable: 20,
+      })
+    );
+    expect(metrics[0].interactionTime).toBeGreaterThanOrEqual(500);
+    expect(metrics[0].interactionTime).toBeLessThanOrEqual(2000);
+  })
+);
+
+test(
+  'emits a metric when the user sorts',
+  setupTest(async ({ page, wrapper, getInteractionMetrics, waitForLoadingFinished }) => {
+    await page.click(wrapper.findColumnSortingArea(2).toSelector());
+    await waitForLoadingFinished();
+
+    const metrics = await getInteractionMetrics();
+    expect(metrics).toHaveLength(1);
+    expect(metrics[0]).toEqual(
+      expect.objectContaining({
+        userAction: 'sorting',
+        componentIdentifier: 'Instances',
+        instanceIdentifier: 'the-instances-table',
+        noOfResourcesInTable: 20,
+      })
+    );
+    expect(metrics[0].interactionTime).toBeGreaterThanOrEqual(500);
+    expect(metrics[0].interactionTime).toBeLessThanOrEqual(2000);
+  })
+);
+
+test(
+  'emits a metric when the user changes preferences',
+  setupTest(async ({ page, wrapper, getInteractionMetrics, waitForLoadingFinished }) => {
+    await page.click(wrapper.findCollectionPreferences().findTriggerButton().toSelector());
+    await delay(4000);
```

Comment:
Just a suggestion, if you were wondering how to speed up this test, I think we could make `USER_ACTION_TIME_LIMIT` configurable for dev pages and set it to something shorter